include/TriangleNodesProvisioner.hpp

```diff
@@ -14,15 +14,40 @@
 #include <memory>
 
 namespace blitzdg {
+    using real_mat_smart_ptr = std::unique_ptr<real_matrix_type>;
+    using real_vec_smart_ptr = std::unique_ptr<real_vector_type>;
+
```

Comment:
I would probably avoid having these using declarations here and instead put them in `Types.h`, or, just keep them local to where you need them.

Reply:
Good idea.